docs/website/docs/dlt-ecosystem/destinations/clickhouse.md

```diff
@@ -104,7 +111,8 @@ Data is loaded into ClickHouse using the most efficient method depending on the
 
 `Clickhouse` does not support multiple datasets in one database, dlt relies on datasets to exist for multiple reasons.
 To make `clickhouse` work with `dlt`, tables generated by `dlt` in your `clickhouse` database will have their name prefixed with the dataset name separated by
-the configurable `dataset_table_separator`. Additionally, a special sentinel table that does not contain any data will be created, so dlt knows which virtual datasets already exist in a
+the configurable `dataset_table_separator`.
+Additionally, a special sentinel table that doesn't contain any data will be created, so dlt knows which virtual datasets already exist in a
```

Comment:
Perhaps out of scope for this PR, but have you considered using pipeline [state](https://dlthub.com/docs/general-usage/state) instead of a sentinel table?

Reply:
That is a really good suggestion!

I've dropped the sentinel table workaround for more canonical usage of dlt:

 https://github.com/dlt-hub/dlt/blob/409487ce0669a06941769c0a5414c8d71289d2b7/dlt/destinations/impl/clickhouse/clickhouse.py#L310-L345

Reply:
@Pipboyguy where is this change? I can't find it in the PR. Overall idea is good but in this very case it is impossible to execture. `dlt` stores state in the very dataset you are creating here. so it cannot be used to check if the dataset was created (it must be in order to use state).

in this specific code you are using package state. the scope of this state is a single load. so you will not be able to refer to it in the next load. some simple tests will pass. but not much else

sentinel table is quite a good idea and let's keep it. I plan to standardize it maybe in some kind of mixin class. We use sentinels in vector databases and dremio and the pattern is always the same.

Reply:
@rudolfix I already reverted back to sentinel table method in latest commits. It is indeed a much simpler a more robust solution imho